active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,565 @@
+| Action Button | * Allows the user to invoke a custom event. <br> * This is the only non-Close button provided out of the box. Other buttons may be implemented in the Content Area. <br><br> ![A teaching tip with a sample title and subtitle. At the bottom of the tip are two buttons, a blue button on the left that reads "Default" and a gray button on the right that reads "Secondary".](ActionButton.PNG) |
+| Pointer | * Triangular extension of the tip body that can be used to indicate that the tip is referring to on-screen UI element. <br> * When PointerMode is set to auto, the pointer will automatically appear on tips that are attached to a target and off for tips that are not attached to a target. <br> * Prefers to center on target. <br> * Maintains a 12px margin from edges of the tip. <br> * Not animated. <br> * Will not have a shadow as shadows cannot yet be added to nonrectangular surfaces. ![An empty teaching tip with a triangular pointer centered outward on its bottom edge.](Pointer.PNG) |
+| Icon | * Added to the left of the Title and Subtitle by default and automatically moved to the right for RTL languages. <br><br> ![A teaching tip with a sample title and subtitle. To the left of the title is a sample icon showing a graph of connected dots.](Icon.PNG) |
+| Hero Content | * Hero Content is media that stretches to the edges of a tip. <br> * Can be placed at the top or bottom of a tip. <br><br> ![A teaching tip with a sample title and subtitle. Above the title and subtitle is a large border-to-border picture of a giraffe.](HeroContent.PNG) |
+| Scroll Bar | * If the tip's contents are large enough to require scrolling, a scrollbar which will not intersect the Close (X) Button will be added to the content area. <br><br> ![A tall teaching tip with a scrollbar on the right a long string of "Lorem ipsum" text in the body.](ScrollBar.PNG) |
+
+### Behavioral Components
+
+ | Property | Notes |
+|:---:|:---|
+| Opening | * A tip is shown by setting its IsOpen property to true. <br> * Tips will animate on opening. <br> * When a tip does not have enough available window space to fully show in any location [see Placement], it will not open and will instead overwrite IsOpen to false. |
+| Closing | * There are three ways to close a tip: set the IsOpen property to false, the user invokes a close button, or the tip is closed via light dismiss. These will return the method used in TeachingTipCloseReason.  <br> * Closing can be deferred by taking a handle to the deferral object in the closing event args. |
```

Comment:
Grammar nitpick: Parallel construction.

> There are three ways a tip can close: The program sets the `IsOpen` property to `false`, the user invokes the Close button, or the tip closes due to light dismiss. Use the `TeachingTipCloseReason` to determine which case has occurred.

Second sentence needs some wordsmithing too, because the deferral doesn't defer closing. It defers the decision whether or not to close.

> Closing can be prevented by setting the `Cancel` property to `true`. You can use a deferral to respond asynchronously to the event.

Reply:
[Fixed](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/f6f1c1d4d93efaeec1d4ac32526925eddc16fdcb)!